WalletWasabi.Tests/UnitTests/Crypto/HashTests.cs

```diff
@@ -0,0 +1,72 @@
+using NBitcoin.Secp256k1;
```

Comment:
`HashTests.cs` -> `HashHelpersTests.cs`?

Reply:
Nah, actually this whole class is a design flaw. Every time you use `Helpers`, `Tools` or other meaningless naming pattern, it means you have a design smell. So I don't want to encourage this bug by solidifying it in tests, too. I however renamed it to `HashingTests`, that may be more expressive.

Reply:
Everything is an antipattern! :-D

Reply:
I don't understand this approach. You say, you don't like `Helpers` & `Tools` but instead of renaming the class to fix this naming issue, you name the test in a way that is not consistent with how other tests are being named. How is this better? If anything, it teaches people to name tests whatever they like.

Reply:
It's not a naming issue. It's a design issue, the naming just suggests that something is not right. Maybe in this specific case, the whole class should be killed as it as the code is tiny here and let the users use the code directly instead, or specify some of the implementations here in extension methods instead. IDK, it's not terribly important IMO.

Why the `Helpers`, `Tools` and `Utils` is an antipattern, because that's an admission that you cannot map your code to the real world:

"I am implementing something. Where should this logic go? IDK, let's just throw it in the `Tools`." This is how you build up technical debt. But technical debt is not inherently bad, it's actually a very useful thing, it's just if you're not careful, it'll grow and paying it back will be extremely expensive in some cases. See `Wallet.cs` in Wasabi or `main.cpp` in Bitcoin Core: it would've been more beneficial getting those right the first time than undergo huge amount of development efforts down the line to save what's savable, to decouple it.

In fact I may even be comfortable to say that maybe recognizing when to build up technical debt and when not to is the only difference between an experienced dev and a rockstar dev.

Anyway, I don't generally like to couple a test to a class, I much prefer categorizing based on functionality and not files, but if the class is failed to begin with, I feel much stronger about this decision.

Reply:
> Why the Helpers, Tools and Utils is an antipattern, because that's an admission that you cannot map your code to the real world:

I don't know. I don't feel so strongly about those names. Naming is hard and I'm not that good at it so I won't judge this.  

> Anyway, I don't generally like to couple a test to a class, I much prefer categorizing based on functionality and not files, but if the class is failed to begin with, I feel much stronger about this decision.

I just think it's good to pick one naming rule for tests and stick with it. Given that I saw many tests with `Tests.cs` suffix, I thought that it's your convention for naming unit tests. 

I can't find any information about that in https://github.com/zkSNACKs/WalletWasabi/blob/master/CONTRIBUTING.md file. Is it defined somewhere?

So in short:

1) If there is a naming rule for tests, I would personally stick with it because it helps other developers and makes their life easier. And if you find the current rule bad, then just propose better naming.
2) If there isn't any such rule, I would try to add one, so others won't have to think about that much and rather spend their time something productive that moves the project forward.

Reply:
> Why the Helpers, Tools and Utils is an antipattern, because that's an admission that you cannot map your code to the real world:

I don't know. I don't feel so strongly about those names. Naming is hard and I'm not that good at it so I won't judge this.  

> Anyway, I don't generally like to couple a test to a class, I much prefer categorizing based on functionality and not files, but if the class is failed to begin with, I feel much stronger about this decision.

I just think it's good to pick one naming rule for tests and stick with it. Given that I saw many tests with `Tests.cs` suffix, I thought that it's your convention for naming unit tests. 

I can't find any information about that in https://github.com/zkSNACKs/WalletWasabi/blob/master/CONTRIBUTING.md file. Is it defined somewhere?

So in short:

1) If there is a naming rule for tests, I would personally stick with it because it helps other developers and makes their life easier. And if you find the current rule bad, then just propose better naming.
2) If there isn't any such rule, I would try to add one, so others won't have to think about that much and rather spend their time something productive that moves the project forward.

Reply:
I don't have a problem with the naming, I have a problem with the design. The naming is perfect, it just suggests design flaws. Design flaws are fine until you pay the technical debt back. But 99% of the technical debt you never have to pay back, so quick progress is preferable, it's just recognizing when you're working on that 1% where you cannot afford design flaws. IMO here you can.

> If you don't like Helper, Utils classes, then just file issue that it should be removed or something - i.e. push for the end state you would be happy with.

The whole idea is to maximize work not done :) I'm happy to have this small mess here, because we'll most likely never have to deal with it again, but even if we do, it's a small mess, cleaning it up takes less time then writing this comment :)

> I just think it's good to pick one naming rule for tests and stick with it. Given that I saw many tests with Tests.cs suffix, I thought that it's your convention for naming unit tests.

IDK how tests should be named. From the first glance the `Tests` suffix seems redundant for me, I just never slowed down to figure it out and adjust my coding patterns for that.

I by no means want to endorse Kevin Henney's grammar nazi approaches to comments and namings of software developers, but he has some really good points how much we are doing is basically cargo cult. I suspect the `Tests` suffix is just as useless as the `Exception` suffix or commenting the obvious: https://www.youtube.com/watch?v=CzJ94TMPcD8